adagrams/game.py

```diff
@@ -1,11 +1,128 @@
-def draw_letters():
-    pass
+import random
+
+def draw_letters(): 
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    #make an empty list that will hold all available letters
+    available_letters = []
+
+    #looping through letter pool dic keys, and adding the key the amount
+    #of times the value designates to available letters list
+    for letter in LETTER_POOL.keys():
+        for num in range(LETTER_POOL[letter]):
+            available_letters.append(letter)
+
+    #choose 10 random letters from available letters
+    chosen_letters = random.sample(available_letters, 10)  
+    return chosen_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    letter_bank_copy = letter_bank.copy()
+    #add a loop that checks if letter in word is in letter bank copy. 
+
+    for letter in word.upper():
+        #if letter in bank copy remove the letter, else return false and break
+        if letter in letter_bank_copy:
+            letter_bank_copy.remove(letter)
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    
+    score = 0
+    score_values = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10 
+    }
+    #sum up each letter in word to score, use .upper to make all caps
+    for letter in word.upper():
+        score+= score_values[letter]
+
+    #add 8 points if len of word is >= 7
+    if len(word) >= 7:
+        score+= 8
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #initialize word_score_dict and highest_score_list
+    word_scores_dict = {}
+    highest_score_list = []
+
+    #add scores as keys to word_score_dict, and words as values in a list
```

Comment:
Nice! I had actually never thought of using the scores as keys, which totally helps with the issue of finding multiple words with the same score.